data/benefits/dynamic/apa.ts

```diff
@@ -0,0 +1,205 @@
+  },
+  departement_seine_saint_denis: {
+    label: "du département de Seine Saint Denis",
+    link: "https://seinesaintdenis.fr/Allocation-Departementale-Personnalisee-d-Autonomie.html",
+    form: "https://seinesaintdenis.fr/IMG/pdf/formulaire_demande_adpa.pdf",
+  },
+  departement_val_de_marne: {
+    label: "du département du Val de Marne",
+    link: "https://www.valdemarne.fr/a-votre-service/personnes-agees/allocation-personnalisee-dautonomie-a-domicile-apad",
+    form: "https://www.valdemarne.fr/download/sites/default/files/formulaires/webformudemande_apa_-2018_.pdf",
+  },
+  intercommunalite_metropole_lyon: {
+    label: "de la Métropole de Lyon",
+    link: "https://www.grandlyon.com/services/allocation-personnalisee-d-autonomie.html",
+    form: "https://www.grandlyon.com/fileadmin/user_upload/media/pdf/pa-ph/personnes-agees/20170802_dossier_demande_apa.pdf",
+  },
+}
+
+function formatBenefit(customizationBenefit, institution) {
+  return {
+    id: `${institution.replace(/_/g, "-")}-apa-eligibilite`,
+    ...DEFAULT_APA,
+    description: `L’allocation personnalisée d’autonomie (APA) ${customizationBenefit.label} est une aide réservée aux plus de
+    60 ans en perte d’autonomie. À travers un plan d’action, elle favorise le
+    maintien à domicile et l’amélioration de la qualité de vie des personnes âgées
+    en établissement en subventionnant des services d’aides à la personne. Sa
+    gestion est confiée aux conseils départementaux...`,
+    conditions: [
+      `Faire évaluer votre perte d’autonomie (classement GIR) à domicile par les
+      services sociaux de votre département.`,
+      "Accepter le plan d’aide proposé par votre département.",
+      "Résider depuis plus de trois mois dans votre département.",
+      'Ne pas percevoir <a target="_blank" rel="noopener" title="Service Public.fr - Peut-on cumuler l’APA avec d’autres revenus ? - Nouvelle fenêtre" href="https://www.service-public.fr/particuliers/vosdroits/F11678">certaines autres aides</a> non cumulables avec l’APA.',
+    ],
+    ...customizationBenefit,
+    label: `Allocation personnalisée d’autonomie ${customizationBenefit.label}`,
+    institution,
+    source: "javascript",
+    conditions_generales: [
+      {
+        type: "age",
+        operator: ">=",
+        value: "60",
+      },
+      {
+        type: "attached_to_institution",
+      },
+    ],
+  }
+}
+
+export function buildAPA(institutionsMap) {
+  const result: any = Object.keys(APA_BY_CODE).reduce(
+    (accum: any, code: string) => {
+      const customizationBenefit = APA_BY_CODE[code]
+
+      const institutionName = institutionsMap[code]?.slug
+      if (!institutionsMap[code] || !institutionName) {
+        console.warn(`No institution for metropole apa ${code}`)
+        return accum
+      }
+      const benefit = formatBenefit(customizationBenefit, institutionName)
+
+      accum.push(benefit)
+      return accum
+    },
+    []
+  )
+  return result
```

Comment:
```suggestion
export function buildAPA() {
  const result: any = Object.keys(APA_BY_CODE).reduce(
    (accum: any, code: string) => {
      const customizationBenefit = APA_BY_CODE[code]
      const benefit = formatBenefit(customizationBenefit, code)

      accum.push(benefit)
      return accum
    },
    []
  )
  return result
```

C'est une suggestion très sale, c'est c'est une façon de dire que ça peut être très simplifié !

Reply:
La clé dans le dictionnaire dans ce fichier est directement le slug de l'institution.

Reply:
@guillett J'ai une suggestion de refacto encore plus simplifiée, je ne trouve pas cela particulièrement sale, c'est plus lisible.

Reply:
Ça ne marche pô.